src/python/pants/backend/codegen/thrift/apache/python/python_thrift_module_mapper.py

```diff
@@ -33,22 +36,27 @@ class PythonThriftMappingMarker(FirstPartyPythonMappingImplMarker):
 @rule(desc="Creating map of Thrift targets to generated Python modules", level=LogLevel.DEBUG)
 async def map_thrift_to_python_modules(
     thrift_targets: AllThriftTargets,
+    python_setup: PythonSetup,
     _: PythonThriftMappingMarker,
 ) -> FirstPartyPythonMappingImpl:
     parsed_files = await MultiGet(
         Get(ParsedThrift, ParsedThriftRequest(tgt[ThriftSourceField])) for tgt in thrift_targets
     )
-    modules_to_providers: DefaultDict[str, list[ModuleProvider]] = defaultdict(list)
+
+    resolves_to_modules_to_providers: dict[ResolveName, DefaultDict[str, list[ModuleProvider]]] = {}
```

Comment:
similar comment here as in the protobuf code re using `defaultdict(lambda: defaultdict(list))`